Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -106,75 +109,122 @@ ROMol *constructMolFromString(const std::string &txt,
     ps.parseName = false;
     ps.sanitize = false;
     ps.removeHs = false;
+    ps.skipCleanup = true;
     mol = SmilesToMol(txt, ps);
   }
   return mol;
 }
 
 }  // end of namespace DaylightParserUtils
 
+namespace {
+void removeSpacesAround(std::string &text, size_t pos) {
+  auto nextp = pos + 1;
+  while (nextp < text.size() && (text[nextp] == ' ' || text[nextp] == '\t')) {
+    text.erase(nextp, 1);
+  }
+  if (pos > 0) {
```

Comment:
Trying with c++ iterators, I don't think it simplifies the code much

```
auto nextp = trim.begin() + pos + 1;
if(pos + 1 < trim.size() && (*nextp == ' ' || *nextp == '\t')) {
 trim.erase(nextp, trim.find_first_not_of(" \t", nextp));
}
```

Reply:
agreed